README.md

```diff
@@ -7,14 +7,170 @@
+production environments.
 
-This repository is versioned by the Odoo version that is being targeted, and
-you can only upgrade one version at a time. For example, to upgrade version
+This repository is tagged by the Odoo version that is being targeted, and
+you can only upgrade one version at a time. Each tag is represented by a
+Dockerfile, prefixed with the tag name.F or example, to upgrade version
 8 to version 10, you need to use the:
 
-* 9.0 branch first
-* 10.0 branch next
+* [9.0 image](./9.0-Dockerfile) first
+* [10.0 imaqe](./10.0-Dockerfile) next
+
+In order to use this image, you must meet the following condition(s):
+
+* Have an existing Dockerized Odoo instance on the version prior to the target version,
+  or the knowledge to make the necessary adjustments to the tutorial.
+
+Basic Instructions
+==================
+
+High level usage of this image is as follows:
+
+* Run the image with the following options:
+    * Mount the root of your current Odoo file store into the container as `/var/lib/odoo_old`
+        * Note that this should not the be `filestore` directory, but the one that contains the
+          `filestore` and `sessions` directories (among others).
+    * Mount your new Odoo file store into the container as `/var/lib/odoo`
+        * You can easily use the old file store path if you wish
+        * Note that this should not the be `filestore` directory, but the one that contains the
+          `filestore` and `sessions` directories (among others).
+    * Link your current PostgreSQL container as `db_old`
+    * Link your new PostgreSQL container as `db`
+        * You can easily use your old container if you wish. This is primarily to facilitate PSQL
+          major version upgrades.
+    * Add environment variables for the PostgreSQL credentials (`PGUSER`, `PGPASSWORD`)
+        * Add environment variables for the old PostgreSQL db. These will default to the new
+          db credentials if not defined. (`PGUSER_OLD`, `PGPASSWORD_OLD`)
+    * Add environment variables for the source and target database names (`DB_SOURCE`, `DB_TARGET`)
+
+Given the above, and assuming the below:
+
+* You have a database container named `postgresql`
+    * There is a user named `odoo_user` with the password `odoo_password` that has superuser
+      access to the database.
+* You have an Odoo v8 database named `odoo_v8`
+* You have an Odoo v8 filesystem on your host at `/var/lib/odoo_v8`
+* You want to upgrade to Odoo v10
+
+You would run the following commands:
+
+```bash
+export PG_USER=odoo_user
+export PG_PASSWORD=odoo_password
+
+# Migrate v8 to v9
+docker run \
+    --link postgresql:db \
+    --link postgresql:db_old \
+    -v /var/lib/odoo_v8:/var/lib/odoo_old \
+    -v /var/lib/odoo_v9:/var/lib/odoo \
+    -e "PGDATABASE=odoo_v9" \
+    -e "PGUSER=${PG_USER}" \
+    -e "PGPASSWORD=${PG_PASSWORD}" \
+    -e "DB_SOURCE=odoo_v8" \
+    -e "DB_TARGET=odoo_v9" \
+    laslabs/odoo-upgrade:9.0
```

Comment:
It's common that when migrating an addon from, say v8 to v9, the migration scripts from versions 7 and older are removed.

So, I guess you'd have to add 2 extra commands here that build the new project and upgrade it.

Reply:
I'm not following? You're meant to step up one version at a time, so I don't think this would be an issue? Assuming upgrading to v9, I would expect the v8=>v9 migration scripts in v9 branch. Am I wrong there?

Reply:
No, I mean the migration scripts in the addon itself, if any.